parachain/pallets/omni-account/src/lib.rs

```diff
@@ -587,6 +669,20 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		fn do_accept_intent(
+			who: T::AccountId,
+			intent_id: IntentId,
+			intent: Intent,
+		) -> DispatchResult {
+			ensure!(intent_id > Self::accepted_intent_ids(&who), Error::<T>::IntentIdTooSmall);
+			ensure!(!Intents::<T>::contains_key(&who, intent_id), Error::<T>::IntentAlreadyExists);
+			// the continuity should have been checked in the worker already
+			AcceptedIntentIds::<T>::insert(&who, intent_id);
+			Intents::<T>::insert(&who, intent_id, intent.clone());
+			Self::deposit_event(Event::IntentAccepted { who, intent_id, intent });
```

Comment:
Does this mean worker should wait for this call finalization before intent execution ?